lms/migrations/versions/396f46318023_events_backfill.py

```diff
@@ -0,0 +1,75 @@
+"""
+Backfill "events" from lti_launches rows.
+
+Revision ID: 396f46318023
+Revises: 42b39684836f
+Create Date: 2022-08-17 15:16:52.785486
+
+"""
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "396f46318023"
+down_revision = "42b39684836f"
+
+
+CUT_OFF_DATE = "2022-08-17T13:00:08.241"
+"""
+Date when we started inserting launches directly on events.
+Rows in lti_launches would already have corresponding event after this date.
+"""
+
+
+def get_configured_launch_type_pk(conn):
+    return conn.execute(
+        "SELECT id from event_type WHERE type = 'configured_launch'"
+    ).fetchone()[0]
+
+
+def upgrade():
+    conn = op.get_bind()
+
+    launch_event_type_id = get_configured_launch_type_pk(conn)
+
+    result = conn.execute(
+        f"""
+        INSERT INTO event
+            (
+                timestamp,
+                type_id,
+                application_instance_id,
+                course_id
+            )
+        SELECT
+                lti_launches.created,
+                {launch_event_type_id},
+                application_instances.id,
+                grouping.id
+        FROM lti_launches
+        JOIN application_instances
+            ON lti_launches.lti_key = application_instances.consumer_key
+        JOIN grouping
```

Comment:
```suggestion
        LEFT OUTER JOIN application_instances
            ON lti_launches.lti_key = application_instances.consumer_key
       LEFT OUTER JOIN grouping
```

:warning: This isn't right (read to the bottom), but it's something like this?

I'm not sure we will have records for all of the courses? The grouping table is newer than the `lti_launches` table.

I'm getting the following counts:

```sql
SELECT COUNT(1) FROM lti_launches
```
13,037,110

```sql
SELECT COUNT(1) FROM lti_launches
JOIN application_instances
    ON lti_launches.lti_key = application_instances.consumer_key
```
13,033,660

```sql
SELECT COUNT(1) FROM lti_launches
JOIN application_instances
    ON lti_launches.lti_key = application_instances.consumer_key
JOIN grouping 
    ON context_id = grouping.lms_id 
    AND grouping.application_instance_id = application_instances.id 
    AND grouping.type = 'course'
```
11,985,175

```sql
SELECT COUNT(1) FROM lti_launches
LEFT OUTER JOIN application_instances
    ON lti_launches.lti_key = application_instances.consumer_key
LEFT OUTER JOIN grouping 
    ON context_id = grouping.lms_id 
    AND grouping.application_instance_id = application_instances.id 
    AND grouping.type = 'course'
 ```
13,041,841! - This is more than the plain count... so this isn't right? How are we getting _more_ rows here?